libbeat/outputs/fileout/file.go

```diff
@@ -39,7 +40,7 @@ func (out *fileOutput) init(config config) error {
 	out.rotator.Path = config.Path
 	out.rotator.Name = config.Filename
 	if out.rotator.Name == "" {
-		out.rotator.Name = config.Index
+		out.rotator.Name = out.beatName
```

Comment:
Currently `index` is treated specially and uses the hack [here](https://github.com/elastic/beats/blob/master/libbeat/outputs/outputs.go#L79). I think we could remove the hack and use the `beatName` in each output as needed. Is that what you mean?